lib/tasks/enterprises.rake

```diff
@@ -1,6 +1,17 @@
 require 'csv'
 
 namespace :ofn do
+  desc 'deletes all orders and associated entities of an enterprise'
+  task :truncate_distributor, [:distributor_id] => :environment do |_task, args|
+    orders = Spree::Order.where(distributor_id: args.distributor_id)
+
+    inventory_units = Spree::InventoryUnit.where(order_id: orders.pluck(:id))
+    inventory_units.delete_all
+
+    ProxyOrder.where(order_id: orders.pluck(:id)).destroy_all
+    orders.destroy_all
```

Comment:
What happens if the orders have payments? Should all this be in a transaction to roll back everything in case one thing fails to delete?

Reply:
To be clear, my only purpose is to just share what I needed now but not to cover all possible cases. IMO this script should evolve as the need arises.